protocol/x/clob/keeper/equity_tier_limit.go

```diff
@@ -117,48 +120,55 @@ func (k Keeper) ValidateSubaccountEquityTierLimitForNewOrder(ctx sdk.Context, or
 		)
 	}
 
-	// Count all the open orders that are on the `MemClob`.
-	equityTierCount := int32(k.MemClob.CountSubaccountOrders(ctx, subaccountId, filter))
-
-	// Include the number of stateful orders that exist outside of the `MemClob`. This includes the number of
-	// untriggered conditional orders and during `DeliverTx` we add the count of to be committed stateful orders
-	// while in `CheckTx` we add the count of uncommitted stateful orders.
+	equityTierCount := int32(0)
 	if order.IsStatefulOrder() {
+		// For stateful orders we get the stateful order count which represents how many long term and
+		// triggered conditional orders exist in state. We add to that all untriggered conditional orders.
+		// If this is `CheckTx` then we must also add the number of uncommitted stateful orders that this validator
+		// is aware of (orders that are part of the mempool but have yet to proposed in a block).
+		equityTierCount = int32(k.GetStatefulOrderCount(ctx, order.OrderId))
 		equityTierCount += int32(k.CountUntriggeredSubaccountOrders(ctx, subaccountId, filter))
-		if lib.IsDeliverTxMode(ctx) {
-			equityTierCount += k.GetToBeCommittedStatefulOrderCount(ctx, order.OrderId)
-		} else {
+		if !lib.IsDeliverTxMode(ctx) {
 			equityTierCount += k.GetUncommittedStatefulOrderCount(ctx, order.OrderId)
-		}
-	}
 
-	if equityTierCount < 0 {
-		if lib.IsDeliverTxMode(ctx) {
-			err = fmt.Errorf(
+			if equityTierCount < 0 {
+				panic(fmt.Errorf(
+					"Expected ValidateSubaccountEquityTierLimitForNewOrder for new order %+v to be >= 0. "+
+						"equityTierCount %d, statefulOrderCount %d, untriggeredSubaccountOrders %d, "+
+						"uncommittedStatefulOrderCount %d.",
+					order,
+					equityTierCount,
+					k.GetStatefulOrderCount(ctx, order.OrderId),
+					k.CountUntriggeredSubaccountOrders(ctx, subaccountId, filter),
+					k.GetUncommittedStatefulOrderCount(ctx, order.OrderId),
+				))
```

Comment:
Small suggestion: formatting.

```suggestion
				panic(
				    fmt.Errorf(
					"Expected ValidateSubaccountEquityTierLimitForNewOrder for new order %+v to be >= 0. "+
						"equityTierCount %d, statefulOrderCount %d, untriggeredSubaccountOrders %d, "+
						"uncommittedStatefulOrderCount %d.",
					order,
					equityTierCount,
					k.GetStatefulOrderCount(ctx, order.OrderId),
					k.CountUntriggeredSubaccountOrders(ctx, subaccountId, filter),
					k.GetUncommittedStatefulOrderCount(ctx, order.OrderId),
				    ),
				)
```